client/app/scripts/superdesk-workspace/workspace.js

```diff
@@ -24,47 +22,29 @@
 
         var PREFERENCE_KEY = 'workspace:active',
             RESOURCE = 'workspaces',
-            deferEdit,
             self = this;
 
-        /**
-         * Start editing of new custom workspace
-         *
-         * @return {Promise}
-         */
-        function create() {
-            self.edited = {user: session.identity._id};
-            deferEdit = $q.defer();
-            return deferEdit.promise;
-        }
-
-        /**
-         * Resolve edit promise with given workspace
-         *
-         * @param {object} workspace
-         * @return {object}
-         */
-        function resolvePromise(workspace) {
-            deferEdit.resolve(workspace);
-            self.edited = null;
-            return workspace;
-        }
-
-        /**
-         * Cancel workspace editing
-         */
-        function cancelEdit() {
-            self.edited = null;
-            deferEdit.reject();
+        function save(workspace) {
+            workspace.user = workspace.user || session.identity._id;
+            return api.save(RESOURCE, workspace).then(updateActive);
         }
 
-        /**
-         * Save currently edited workspace
-         *
-         * @return {Promise}
-         */
-        function save() {
-            return api.save(RESOURCE, self.edited).then(updateActive).then(resolvePromise);
+        function _delete(workspace) {
+            return api.remove(workspace)
+            .then(function() {
+                if (self.active && self.active._id === workspace._id) {
+                    return self.queryUserWorkspaces()
+                    .then(function(items) {
+                        if (items && items.length) {
+                            self.setActive(items[0]);
+                        } else {
+                            self.setActive(null);
+                        }
+                        self.getActive();
+                    });
+                }
+                return $q.when();
+            });
```

Comment:
Could this nested `.then()` be flattened perhaps?

I would also suggest to reverse the condition for readability so that the main block in the function is not indented for an extra level, e.g.:

```js
...
.then(function() {
    if (!self.active | | self.active._id !== workspace._id) {
        return $q.when();
    }

    // the main block, now outside of the `if` block
});
```